crates/solver/src/liquidity/order_converter.rs

```diff
@@ -76,16 +76,22 @@ impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
     fn encode(&self, executed_amount: U256, encoder: &mut SettlementEncoder) -> Result<()> {
         let is_native_token_buy_order = self.order.order_creation.buy_token == BUY_ETH_ADDRESS;
 
-        if is_native_token_buy_order {
+        if !self.is_liquidity_order && is_native_token_buy_order {
```

Comment:
Why do we not add token equivalency for liquidity orders? They still need a `0xeee..eee` price if they are buying ETH right?

Reply:
**edit**: I understand why its not now from the code below (when the additional prices are added to the price vector for liquidity orders).

Reply:
Good point. Not obvious, hence I added a comment: https://github.com/gnosis/gp-v2-services/pull/1544/commits/59c699363dbdc808a7c1136a96a23a5d282d82ed